src/libs/parsers/tests/formatDates.test.ts

```diff
@@ -8,86 +8,368 @@ import {
   `('formatDates($dates) \t= $expected', ({ dates, expected }) => {
     const timestampsInSeconds = dates && dates.map((date: Date) => date.valueOf())
-    expect(formatDates(timestampsInSeconds)).toBe(expected)
+    expect(formatDates(timestampsInSeconds)).toEqual(expected)
   })
 })
 
 describe('formatToFrenchDate', () => {
   beforeAll(() => {
-    mockdate.set(Nov1)
+    mockdate.set(NOVEMBER_1_2020)
   })
 
   it.each`
     date                                        | expected
-    ${Nov12}                                    | ${'12 novembre 2020'}
-    ${Oct5}                                     | ${'5 octobre 2020'}
-    ${Oct5.valueOf()}                           | ${'5 octobre 2020'}
+    ${NOVEMBER_12_2020}                         | ${'12 novembre 2020'}
+    ${OCTOBER_5_2020}                           | ${'5 octobre 2020'}
+    ${OCTOBER_5_2020.valueOf()}                 | ${'5 octobre 2020'}
     ${{ day: 5, month: 'octobre', year: 2020 }} | ${'5 octobre 2020'}
     ${'2030-02-05T00:00:00Z'}                   | ${'5 février 2030'}
-    ${Nov12.toString()}                         | ${'12 novembre 2020'}
+    ${NOVEMBER_12_2020.toString()}              | ${'12 novembre 2020'}
   `('formatToFrenchDate($dates) \t= $expected', ({ date, expected }) => {
-    expect(formatToFrenchDate(date)).toBe(expected)
+    expect(formatToFrenchDate(date)).toEqual(expected)
   })
 })
 
 describe('formatDatePeriod', () => {
-  beforeAll(() => {
-    mockdate.set(Nov1)
+  it('should return a formatted string date for a period of one day', () => {
+    expect(
+      formatDatePeriod([
+        new Date('2023-04-10T10:00:00Z').getTime(),
+        new Date('2023-04-10T23:59:59Z').getTime(),
+      ])
+    ).toEqual('10 avril 2023')
   })
-  it.each`
-    dates                           | expected
-    ${[]}                           | ${undefined}
-    ${[Nov12]}                      | ${'12 novembre 2020'}
-    ${[Nov12, Nov12]}               | ${'12 novembre 2020'}
-    ${[Nov20Morning, Nov20Evening]} | ${'20 novembre 2020'}
-    ${[Dec5]}                       | ${'5 décembre 2020'}
-    ${[Nov12, Nov12]}               | ${'12 novembre 2020'}
-    ${[Nov1, Nov12, Oct5]}          | ${'Du 1 au 12 novembre 2020'}
-    ${[Dec5, Nov12]}                | ${'Du 12 novembre au 5 décembre 2020'}
-    ${[Jan2021, Nov12]}             | ${'Du 12 novembre 2020 au 5 janvier 2021'}
-  `('formatDatePeriod($dates) \t= $expected', ({ dates, expected }) => {
-    expect(formatDatePeriod(dates)).toBe(expected)
+
+  it('should return a formatted string date for a period of multiple days in the same month and year', () => {
+    expect(
+      formatDatePeriod([
+        new Date('2023-04-10T10:00:00Z').getTime(),
+        new Date('2023-04-15T23:59:59Z').getTime(),
+      ])
+    ).toEqual('Du 10 au 15 avril 2023')
+  })
+
+  it('should return a formatted string date for a period of multiple days in different months and the same year', () => {
```

Comment:
`should return a formatted string date for a period of multiple days in different months of the same year`